### PR TITLE
terraform: Check for sensitive values in outputs

### PR DESCRIPTION
**Pull request by alisdair**
Sensitive values may not be used in outputs which are not also marked as sensitive. This includes values nested within complex structures.

Note that sensitive values are unmarked before writing to state. This means that sensitive values used in module outputs will have the sensitive mark removed. At the moment, we have not implemented sensitivity propagation from module outputs back to value marks.

This commit also reworks the tests for `NodeApplyableOutput` to cover more existing behaviour, as well as this change.

### CODE REVIEW

File: terraform/node_output_test.go [hunk 1]
```diff
@@ -1,61 +1,142 @@
...
-	}{
-		{
-			// Eval should recognize a single map in a slice, and collapse it
-			// into the map value
-			"single-map",
-			cty.MapVal(map[string]cty.Value{
-				"a": cty.StringVal("b"),
-			}),
-			false,
-		},
```
Comment (alisdair) [inline terraform/node_output_test.go:29]: This test case has been collapsed into `TestNodeApplyableOutputExecute_knownValue` and expanded a bit.

File: terraform/node_output_test.go [hunk 2]
```diff
@@ -1,61 +1,142 @@
...
-			"multi-map",
-			cty.ListVal([]cty.Value{
-				cty.MapVal(map[string]cty.Value{
-					"a": cty.StringVal("b"),
-				}),
-				cty.MapVal(map[string]cty.Value{
-					"c": cty.StringVal("d"),
-				}),
-			}),
-			true,
```
Comment (alisdair) [inline terraform/node_output_test.go:41]: I'm not sure what this was intending to test, but it doesn't error and the previous test didn't verify that it did. Removed and added `TestNodeApplyableOutputExecute_invalidDependsOn` to cover an existing returned error case.

**Review by alisdair — COMMENTED** [review]
📝 on the test changes

**Review by pselle — APPROVED** [review]
Thanks for including a note here that we're not yet following sensitivity on outputs once they're written, this is a good caveat to consider!

**Pull request merged**